src/hal/src/command/raw.rs

```diff
@@ -526,3 +526,9 @@ pub trait RawCommandBuffer<B: Backend>: Any + Send + Sync {
         T: 'a + Borrow<B::CommandBuffer>,
         I: IntoIterator<Item = &'a T>;
 }
+
+/// A trait for types that can be converted into raw command buffer.
+pub trait IntoRawCommandBuffer<B: Backend, C> {
```

Comment:
Did you consider trying the standard `Into` instead of rolling out a new trait?

Reply:
why do you need the `Capability` bound on the trait anyway?

Reply:
Good point. The unsafety story is rather poor, still, but this simplification doesn't worth it. Let's proceed with your trait and `C` then.